pwiz_tools/Skyline/TestUtil/TestFunctional.cs

```diff
@@ -286,18 +287,32 @@ protected static TDlg ShowDialog<TDlg>(Action act, int millis = -1) where TDlg :
                 dlg = WaitForOpenForm<TDlg>(millis);
             Assert.IsNotNull(dlg);
 
+            return dlg;
+        }
+
+        private static void EnsureScreenshotIcon(Form dlg)
+        {
             // Making sure if the form has a visible icon it's Skyline release icon, not daily one.
             if (IsRecordingScreenShots && dlg.ShowIcon && !ReferenceEquals(dlg, SkylineWindow))
             {
                 if (dlg.FormBorderStyle != FormBorderStyle.FixedDialog ||
-                    dlg.Icon.Handle == Resources.Skyline.Handle)    // Normally a fixed dialog will not have the Skyline icon handle
+                    AreIconsEqual(dlg.Icon, Resources.Skyline))    // Normally a fixed dialog will not have the Skyline icon
                 {
-                    var ico = dlg.Icon.Handle;
-                    if (ico != SkylineWindow.Icon.Handle)
+                    if (!AreIconsEqual(dlg.Icon, SkylineWindow.Icon))
                         RunUI(() => dlg.Icon = SkylineWindow.Icon);
                 }
             }
-            return dlg;
+        }
+
+        private static bool AreIconsEqual(Icon icon1, Icon icon2)
+        {
+            using var ms1 = new MemoryStream();
+            using var ms2 = new MemoryStream();
+
+            icon1.Save(ms1);
+            icon2.Save(ms2);
+
+            return StructuralComparisons.StructuralEqualityComparer.Equals(ms1.ToArray(), ms2.ToArray());
```

Comment:
I usually use "SequenceEqual" as in:
`ms1.ToArray().SequenceEqual(ms2.ToArray())`